api/src/services/submission-service.ts

```diff
@@ -131,4 +142,84 @@ export class SubmissionService extends DBService {
   async listSubmissionRecords(): Promise<ISubmissionModelWithStatus[]> {
     return this.submissionRepository.listSubmissionRecords();
   }
+
+  /**
+   *
+   *
+   * @param {number} submissionId
+   * @return {*}  {Promise<ISourceTransformModel['metadata_transform_precompile']>}
+   * @memberof SubmissionService
+   */
+  async getEMLStyleSheetKey(submissionId: number): Promise<ISourceTransformModel['transform_precompile_key']> {
+    return (await this.submissionRepository.getSourceTransformIdBySubmissionId(submissionId)).transform_precompile_key;
+  }
+
+  /**
+   *
+   *
+   * @param {number} submissionId
+   * @return {*}  {Promise<any>}
+   * @memberof SubmissionService
+   */
+  async getStylesheetFromS3(submissionId: number): Promise<any> {
+    try {
+      const stylesheet_key = await this.getEMLStyleSheetKey(submissionId);
+
+      if (!stylesheet_key) {
+        throw new ApiExecuteSQLError('Failed to retrieve stylesheet key', [
+          'SubmissionRepository->getStyleSheetKey',
+          'stylesheet_key was null'
+        ]);
+      }
+
+      const s3File = await getFileFromS3(stylesheet_key);
+
+      if (!s3File) {
+        throw new HTTP500('Failed to get file from S3');
+      }
+      return s3File;
+    } catch (error) {
+      defaultLog.error({ label: 'getS3File', message: 'error', error });
```

Comment:
nit: can you change `getS3File` to the name of this function

Reply:
Also, possibly consider just removing the try catch altogether since we just re-throw the error anyways.

Reply:
yes, true, we do log it higher up.